### PR TITLE
add support for build.privileged and service.cgroup

### PR DESCRIPTION
**Pull request by ndeloof**

### CODE REVIEW

File: types/types.go [hunk 1]
```diff
@@ -354,12 +356,12 @@ type ThrottleDevice struct {
 // ShellCommand is a string or list of string args.
 //
 // When marshaled to YAML, nil command fields will be omitted if `omitempty`
-// is specified as a struct tag. Explicitly empty commands (i.e. `[]` or `''`)
+// is specified as a struct tag. Explicitly empty commands (i.e. `[]` or `”`)
```
Comment (glours) [inline types/types.go:359]: @ndeloof you're IDE again?
```suggestion
// is specified as a struct tag. Explicitly empty commands (i.e. `[]` or `''`)
```

File: types/types.go [hunk 2]
```diff
@@ -354,12 +356,12 @@ type ThrottleDevice struct {
...
 // will serialize to an empty array (`[]`).
 //
 // When marshaled to JSON, the `omitempty` struct must NOT be specified.
 // If the command field is nil, it will be serialized as `null`.
-// Explicitly empty commands (i.e. `[]` or `''`) will serialize to an empty
+// Explicitly empty commands (i.e. `[]` or `”`) will serialize to an empty
```
Comment (glours) [inline types/types.go:364]: same?
```suggestion
// Explicitly empty commands (i.e. `[]` or `''`) will serialize to an empty
```

**Review by glours — CHANGES_REQUESTED** [review]

File: types/types.go (hunk 1 above)
Comment (ndeloof) [reply types/types.go:359]: 😓 I thought I found the right option to prevent this.

**Review by glours — APPROVED** [review]
LGTM

**Pull request merged**